### ISSUE TITLE
Flair Pretrained NER-English-Large Model Export to ONNX

### ISSUE THREAD

**Issue by chouksep** [flairNLP/flair#2625]
Hi, I am working on exporting flair ner-english-large model into ONNX format.  I am trying a couple of techniques which includes implementing pytorch function **torch.onnx.export(model, dummy_input)** and **using hugging face transformer utilities for onnx conversion**. I am facing following issues while implementing these techniques. 

1. 
Input: 
```
model = SequenceTagger.load("flair/ner-english-large")
shape_of_first_layer =list(model.parameters())[0].shape
N,C = shape_of_first_layer[:2]
dummy_input = torch.Tensor(N,C)
torch.onnx.export(model, dummy_input,'trial.onnx',                                
                    opset_version=11,                            
                    input_names=['input'],
                    output_names=['output'] )

```
**output:** 
![image](https://user-images.githubusercontent.com/99215109/152863450-9c22f65e-cb85-451e-9d33-f3fd394f78d4.png)

2. Input:
```
model = SequenceTagger.load("flair/ner-english-large")
embeddings = TransformerWordEmbeddings(
    model='xlm-roberta-large',
    layers="-1",
    subtoken_pooling="first",
    fine_tune=True,
    use_context=True,
)
with torch.no_grad():
    my_sent ="Replace me by any text you'd like."
    sentence = Sentence(my_sent)
    inputs=embeddings.embed(sentence)
    model(inputs)
    torch.onnx.export(model,                                            
                    inputs,                  
                    'trial.onnx',                                
                    opset_version=11,
                    input_names=['input'], 
                    output_names=['output']        
                    )
```
Output:
![image](https://user-images.githubusercontent.com/99215109/152864336-d1863b79-37f7-4e1e-81c9-d18265ff8280.png)

3. I also tried to use hugging face transformer utilities for onnx conversion : https://huggingface.co/docs/transformers/serialization But I noticed model card for ner-english-large in the hugging face model zoo does not have config.json file. And when I apply this technique, it throws following error:
![image](https://user-images.githubusercontent.com/99215109/152864991-ac186f29-52c6-4b39-9100-03c9001696b4.png)

I would like to know how I can solve any of these issues in order to successfully convert/export flair model into onnx. Also, I would like to know if there is any alternative way of doing this. 

Thanks in advance!

**Comment by lishengfever** [comment]
Is this problem solved?

**Comment by chouksep** [comment]
> Is this problem solved?
Not yet.

**Comment by helpmefindaname** [comment]
Hello @chouksep,
I don't think using ONNX out of the box is possible right now.
This would require some redesign of the model class and it's implementations (make the `forward` method of the model to use raw tensors instead of the specific `DataPoint`s, split up the logic around the NN (e.g. decoding the crf outputs, creating `Label` objects and link them to the `DataPoints`) in a way, that you don't need to create a pytorch model, figure out a way to handle embeddings, ...)

However, you can try to convert the transformer embeddings, using the in-memory method: https://huggingface.co/docs/transformers/serialization#exporting-the-model you can get the HF-Model via `sequence_tagger.embeddings.model` and the HF-Tokenizer via `sequence_tagger.embeddings.tokenizer`
Then you would have to implement a OnnxTransformerEmbedding class, that would be similar to [this class](https://github.com/flairNLP/flair/blob/master/flair/embeddings/base.py#L169), but creates/uses the ONNX model instead of the transformer model (replace all code occurences of `self.model`)

That would still be a hard task, but I think it could be doable for someone who spends enough time.

**Comment by helpmefindaname** [comment]
Hello,
I made a PR which should make it possible to export transformer embeddings to onnx can you check out https://github.com/flairNLP/flair/pull/2739/files#diff-12961c27a5afbdccd88c22a4678b771edbe9a64e512fd5db7b3ac4df4fa4234b and see if that works?
